system/src/system_network_internal.h

```diff
@@ -599,34 +613,49 @@ class ManagedNetworkInterface : public NetworkInterface
         return (SPARK_WLAN_STARTED && WLAN_CONNECTING);
     }
 
-    void on() override
+    int on() override
     {
         LOG_NETWORK_STATE();
-        if (!SPARK_WLAN_STARTED)
+        if (!SPARK_WLAN_STARTED || !WLAN_INITIALIZED)
         {
+            bool startedPowerOn = !SPARK_WLAN_STARTED;
+            WLAN_INITIALIZED = 0;
             const auto diag = NetworkDiagnostics::instance();
             diag->status(NetworkDiagnostics::TURNING_ON);
-            system_notify_event(network_status, network_status_powering_on);
+            if (startedPowerOn) {
+                // Suppress these events
+                system_notify_event(network_status, network_status_powering_on);
+            }
```

Comment:
Nit: The reverse logic in "Suppress these events" makes my brain hurt.  Can we just omit this comment?